angr/state_plugins/sorted_collection.py

```diff
@@ -0,0 +1,311 @@
+from bisect import bisect_left, bisect_right
+
+class SortedCollection(object):
```

Comment:
This comes from https://code.activestate.com/recipes/577197-sortedcollection/. The license is MIT. We should probably add the LICENSE to the header of the file at least.

Same question as before--should I reproduce this license elsewhere as well?

Reply:
I would prefer enormously to not vendor other people's code, but instead rely on pip-installable dependendies. We have a dep on `sortedcontainers`, which provides tools that do this exactly.

Reply:
RangeTree too. If it's 1000% necessary to have these things vendored please put them in angr/misc.

Reply:
holy fuck you've added literally so much containers code. is this really necessary? has nobody put up a package for this stuff on pypi?

Reply:
I'll put it on my TODO. I agree, this all shouldn't be here.

Reply:
Removed the dependency on sorted_collection.py (though I have to verify the changes work, specifically for the merge code which is currently not working due to #1603 ).

Lots of the other code was unused afaict:

- unpaged_memory.py
- range_map.py
- range_tree.py

Added a TODO above to move pitree into its own package

Reply:
Hrm, on second though, I used SortedList when I probably should've used SortedDict.